src/nfts/saga.ts

```diff
@@ -0,0 +1,52 @@
+import { call, put, select, spawn, takeLeading } from 'redux-saga/effects'
+import { fetchNfts, fetchNftsCompleted, fetchNftsFailed } from 'src/nfts/slice'
+import { getFeatureGate } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import Logger from 'src/utils/Logger'
+import { safely } from 'src/utils/safely'
+import { Actions } from 'src/web3/actions'
+import networkConfig from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+const TAG = 'NftsSaga'
+
+export function* handleFetchNfts() {
+  const showNftsInApp = getFeatureGate(StatsigFeatureGates.SHOW_IN_APP_NFT_GALLERY)
+  if (!showNftsInApp) {
+    Logger.debug(TAG, 'Feature gate not enabled, skipping NFTs list fetch')
+    return
+  }
+
+  const address = yield select(walletAddressSelector)
+  if (!address) {
+    Logger.debug(TAG, 'Wallet address not found, skipping NFTs list fetch')
+    return
+  }
+
+  const url = `${networkConfig.getNftsByOwnerAddressUrl}?address=${address}`
+
+  try {
+    const response = yield call(fetch, url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    })
+    if (!response.ok) yield put(fetchNftsFailed({ error: 'Could not fetch NFTs' }))
```

Comment:
Shouldn't we throw here? And the `catch` will handle it.
Maybe we can include some data from the failed response in the message. I think `response.status` and `response.statusText` are good to have.

Also, it's a good practice to always include the `{ }`.

Reply:
The structure was recommended in https://github.com/valora-inc/wallet/pull/3908#discussion_r1239871536.

Reply:
i think you should throw here, it will land in the catch block so you don't need to duplicate the error handling (calling `fetchNftsFailed` twice)

Reply:
Taking a look at this, wouldn't this make the error more generic? We wouldn't have a separate error for non ok response and a parse error. I am fine with this as long as we're ok losing this error granularity. @dievazqu